package/MDAnalysis/transformations/translate.py

```diff
@@ -63,8 +62,80 @@ def wrapped(ts):
             v = np.float32(vector)
```

Comment:
This is kind of out of scope for this PR, but could be fixed anyway: the test on the vector size and the type conversion could be done once in the parent function so that the wrapped function can just do the translation.

Reply:
Also, it uses `len` on an array; and does not complain for too long vectors.